packages/expo-modules-core/ios/Swift/AppContext.swift

```diff
@@ -158,7 +299,37 @@ public final class AppContext {
    */
   deinit {
     NotificationCenter.default.removeObserver(self)
-    moduleRegistry.post(event: .appContextDestroys)
+
+    // Post an event to the registry only if it was already created.
+    // If we let it to lazy-load here, that would crash since the module registry
+    // has a weak reference the app context which is being deallocated.
```

Comment:
```suggestion
    // Post an event to the registry only if it was already created.
    // If we let it to lazy-load here, that would crash since the module registry
    // has a weak reference to the app context which is being deallocated.
```